### PR TITLE
feat: add entitymatching and delete tags

### PR DESCRIPTION
**Pull request by cowan-macady**
implement ENG-4427

- add entitymatching
- add deletenodetags
- deprecate IngestRecord

### CODE REVIEW

**Review by Septimus4 — CHANGES_REQUESTED** [review]

File: indykite_sdk/ingest/ingest_record.py [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+        warnings.warn(
+            f"{func.__name__}() is deprecated and will be removed in a future version.",
+            DeprecationWarning,
+            stacklevel=2
+        )
+        return func(*args, **kwargs)
+    return wrapper
+
+
+@deprecated
```
[the remark this answers is not in the document]
Comment (cowan-macady) [reply indykite_sdk/ingest/ingest_record.py:27]: it did not work in utils. I will add the warning in the function

[the remark this answers is not in the document]
Comment (cowan-macady) [reply indykite_sdk/ingest/ingest_record.py]: updated

File: tests/test_ingest_batch.py
[the remark this answers is not in the document]
Comment (cowan-macady) [reply tests/test_ingest_batch.py:1]: added

File: indykite_sdk/model/entity_matching_pipeline_config.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        self.similarity_score_cutoff = similarity_score_cutoff
+        self.property_mapping_status = property_mapping_status
+        self.property_mapping_message = property_mapping_message
+        self.entity_matching_status = entity_matching_status
+        self.entity_matching_message = entity_matching_message
+        self.property_mappings = property_mappings
+        self.rerun_interval = rerun_interval
+        self.last_run_time = last_run_time
+        self.report_url = report_url
+        self.report_type = report_type
```
[the remark this answers is not in the document]
Comment (cowan-macady) [reply indykite_sdk/model/entity_matching_pipeline_config.py]: changed for this one

**Review by Septimus4 — APPROVED** [review]
Great, it looks a lot better, and it will be a lot easier to maintain. Hoping we slowly refactor the rest, and make sure next PRs enforce the same rules

**Comment by cowan-macady** [comment]
I added some more refactoring. Will continue next pr

**Pull request merged**